### ISSUE TITLE
test -ot, -nt, and -ef are not flagged as non-POSIX

### ISSUE THREAD

**Issue by engber** [koalaman/shellcheck#3355]
The following line was not flagged:

    if [ "${dst}" -ot "${file}" ]; then

I think that -ot, -nt, and -ef are not POSIX. Linux and BSD seem to support them.

---

#### For bugs
- Rule Id (if any, e.g. SC1000): SC3057, SC2295, SC3011
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

# Usage: schlepNeeded src_html_file dst_html_file [-v var | -v !var]
#   src_html_file: the file being schlepped.
#   dst_html_file: result of a previous schlepBundle.
#   var: conditional code inclusion variables - e.g. #if SOME_VAR ... #endif.
#
# This command determines if a html file needs reprocessing by schlepBundle.
#   
# It compares the modification date src_html_file and all the files it includes
# to the modification date of dst_html_file.
#
# If any of the source files are newer - i.e. destination is out-of-date
# it writes 1 to standard output and the exit status is 0.
#
# If none of the source files are newer - i.e. destination is up-to-date
# it writes 0 to standard output and the exit status is 1.
#
# Note: Even though an output file is not generated, the code inclusion variables
# are still needed because they might affect this list of included files. 
#
# EXIT STATUS
#   0 Any of the source files are newer - i.e. destination is out-of-date.
#   1 None of the source files are newer - i.e. destination is up-to-date.
#
# EXAMPLES
#
# Using the exit status and ignoring standard output is similar to how one
# might use the expr utility.
#
#   if schlepNeeded "${1}" "${2}" >/dev/null; then
#       schlepBundle "${1}" -o "${2}"
#   fi
#
# Alternately, you can use command substitution and compare to 0 or 1.
#
#   if [ $(schlepNeeded "${1}" "${2}") = 1 ]; then
#       schlepBundle "${1}" -o "${2}"
#   fi

scriptDir=$(dirname "${0}")
srcRootPath=${1}
shift
dstRootPath=${1}
shift
srcRootDir=$(dirname "${srcRootPath}")
srcRootName=$(basename "${srcRootPath}")
dstRootDir=$(dirname "${dstRootPath}")
dstRelativeDir="${dstRootDir}/relative"
dstAbsoluteDir="${dstRootDir}/absolute"

[ -z "${srcRootPath}" ] && exit

# Remember to pass the variable args (${@}). If the html file takes advantage
## of preprocessing it may affect the result of --list-only.
fileList=$("${scriptDir}/schlep" --list-only "${@}" "${srcRootPath}")

# schlep --list-only returns a list of newline separated file paths to iterate.
newline='
'

while IFS=$newline read -r file; do

    if [ -d "${dstRootPath}" ]; then

        if [ "${file}" = "${srcRootPath}" ]; then
            dst=${dstRelativeDir}/${srcRootName}
        elif [ "${file:0:1}" = "/" ]; then
            dst=${dstAbsoluteDir}${file}
        else
            # No slash inserted.
            # srcRootDir does _not_ end with slash, so leading slash in file is _not_ stripped.
            dst=${dstRelativeDir}${file#${srcRootDir}}
        fi
    else
        dst=${dstRootPath}
    fi

    echo "src='${file}'"
    echo "dst='${dst}'"

    if [ "${dst}" -ot "${file}" ]; then
        echo 1
        exit 0
    fi

done <<< "$fileList"

echo 0
exit 1
```

#### Here's what shellcheck currently says:
```
Line 68	SC3057: In POSIX sh, string indexing is undefined.
Line 73	SC2295: Expansions inside ${..} need to be quoted separately, otherwise they match as patterns.
Line 87	SC3011: In POSIX sh, here-strings are undefined.
```
#### Here's what I wanted or expected to see:

TODO: Describe expected/desired output

**Comment by e-kwsm** [comment]
The operators are defined in POSIX.1-2024.
<https://pubs.opengroup.org/onlinepubs/9799919799/utilities/test.html>

**Comment by engber** [comment]
I actually looked at that web page and I missed them. Doh!
